docs/examples/shading/plot_martinez_shade_loss.py

```diff
@@ -0,0 +1,240 @@
+"""
+Modelling irradiance loss due to of direct irradiance in modules with bypass diodes
+===================================================================================
+"""  # noqa: E501
+
+# %%
+# This example illustrates how to use the proposed by Martinez et al. [1]_.
+# The model adjusts the incident direct and circumsolar beam irradiance of a
+# PV module based on the number
+# of shaded *blocks*. A *block* is defined as a group of cells protected by a
+# bypass diode. More information on the *blocks* can be found in the original
+# paper [1]_ and in :py:func:`pvlib.shading.direct_martinez`
+# documentation.
+#
+# The following key functions are used in this example:
+#
+# 1. :py:func:`pvlib.shading.direct_martinez` to calculate the adjustment
+#    factor for the output power.
```

Comment:
Initally, we say that the model adjusts the irradiance, but here, we say that it adjusts the power output.